### ISSUE TITLE
Events not binding when included in another component

### ISSUE THREAD

**Issue by Joeao** [marko-js/marko#782]
### Context
If a component includes another component, events will break the component

```javascript
// modal.marko
class {
    destroy() {
        this.el.destroy();
    }
}

<div class="modal-dialog">
    <div class="modal-content ">
        <include(input.body)/>
    </div>
</div>
```
```javascript
// modalContents.marko
class {
    userSettings(e, el) {
    	// do something
    }
}

<include('../modal.marko')>
    <@body>
	<li class="link" on-click('userSettings')>
	    <p>User Settings</p>
	</li>
    </@body>
</include>
```
### Expected Behavior
Component will render with the included component

### Actual Behavior
Getting `Uncaught ReferenceError: __component is not defined` @ `onclick: __component.d("userSettings")`.

Removing the on-click binding makes the error disappear.

<details><summary>Additional Info</summary>

### Your Environment

* Version used: 4.4.21
* Environment name and version (e.g. Chrome 39, node.js 5.4): Chromium 59, Node 6.11.1
* Operating System and version (desktop or mobile): Ubuntu 16.04
* Link to your project: https://github.com/Joeao/marko-starter-demo

### Steps to Reproduce

1. Clone https://github.com/Joeao/marko-starter-demo
2. Run demo
</details>

**Comment by DylanPiercey** [comment]
@Joeao I believe I have also run into this issue.
As a temporary workaround you can nest the outer component in a div like so:

```js
<div>
  <include('../modal.marko')>
      <@body>
        <li class="link" on-click('userSettings')>
          <p>User Settings</p>
        </li>
      </@body>
  </include>
</div>
```

**Comment by Joeao** [comment]
Yeah I've spoken about it briefly with the devs in Gitter, they did request I open this as a separate issue, however it sounds like whatever fixes #780 will fix this too - something to do with no root node to bind to, which shows how your workaround is a good one.

The component I was including wasn't big so I just copied it into each component that needed it.

**Comment by mlrawlings** [comment]
This should be fixed in the latest beta version (`npm i marko@beta`) which will go stable very soon.
